lib/portlayer/exec/container.go

```diff
@@ -451,8 +452,21 @@ func (c *Container) Remove(ctx context.Context, sess *session.Session) error {
 		return c.vm.DeleteExceptDisks(ctx)
 	})
 	if err != nil {
-		c.State = existingState
-		return err
+		f, ok := err.(types.HasFault)
+		if !ok {
+			c.State = existingState
+			return err
+		}
+		switch f.Fault().(type) {
+		case *types.InvalidState:
+			log.Warnf("container VM is in invalid state, unregistering")
+			if err := c.vm.Unregister(ctx); err != nil {
+				log.Errorf("Error while attempting to unregister container VM: %s", err)
+				return err
```

Comment:
If we see an error here can we retry the delete now c.State == StateRemoving? I don't want us to end up with a zombie container where we cannot retry the operation.